libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -615,7 +608,7 @@ uint32_t AP_Frsky_Telem::calc_param(void)
  */
 uint32_t AP_Frsky_Telem::calc_gps_latlng(bool *send_latitude)
 {
-    uint32_t latlng;
+    uint32_t latlng = 0;
```

Comment:
You could do without this with early-return, something like this:

```
uint32_t AP_Frsky_Telem::calc_gps_latlng(bool *send_latitude)
{
    const Location &loc = AP::gps().location(0); // use the first gps instance (same as in send_mavlink_gps_raw)

    // alternate between latitude and longitude
    if ((*send_latitude) == true) {
        (*send_latitude) = false;  // send longitude next time
        if (loc.lat < 0) {
            return ((labs(loc.lat)/100)*6) | 0x40000000;
        }
        return (labs(loc.lat)/100)*6;
    }

    (*send_latitude) = true;  // send latitude next time

    if (loc.lng < 0) {
        return ((labs(loc.lng)/100)*6) | 0xC0000000;
    }
    return ((labs(loc.lng)/100)*6) | 0x80000000;
}
```

Reply:
Also, that parameter could be a reference rather than a pointer.

Reply:
Could we consider using the canonical aircraft position rather than GPS at some point?

That would mean that if you're using beacons for positioning you will get the data from those, for example.

Reply:
this is code I did not write nor touch except for the initialization, it's been this way for ages, I reckon it can be rewritten in a more elegant form as with most of the stuff in this library, it probbaly needs a global refactor

ok for the reference

Do we use the canonical position elsewhere?